### PR TITLE
Add deno port

### PR DESCRIPTION
**Pull request by roziscoding**
This adds a `mod.ts` file, pointing to a deno-compatible version.
I have future plans on merging the two implementations by abstracting the differences between them but, untill I get to it, this will suffice.

### CODE REVIEW

File: mod.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import deno from './src/deno.ts'
+
+export default deno
```
Comment (khaosdoctor) [inline mod.ts:3]: Just something here. If you export it as deno, the auto import will import it as `deno` on the other side. I suggest keeping the same name

Comment (roziscoding) [reply mod.ts:3]: I'm exporting it as default
It'll be imported as `import sugar-env from 'https://...'`

**Review by rwillians — COMMENTED** [review]
Is there a way to extract the main portion of this code so it can be reused between node’s implementation and deno’s?

That way we avoid duplicating code and we might get away with testing it in deno for now

**Comment by roziscoding** [comment]
I thought about that too.
There is, though it's going to require abstracting things a little bit more, like `get`, `_getAs` and `has` becoming factories that receive the environment object, so we can call them in both node and deno exports,  passing the respective object (`process.env` for node and `Deno.env.toObject()` for deno).

Also, `current` would need to be defined as a function too, but it's result could be statically exported from the entrypoints.

If that approach is ok with you, @rwillians, I can work on implementing it tonight.

**Comment by roziscoding** [comment]
Maybe it's a good time for a v2?

**Comment by rwillians** [comment]
> Maybe it's a good time for a v2?

I’d like to keep the exposed API exact the same for now. As long as we keep that, there’s no need to bump a major. We could start a thread to change the API tho if you or @khaosdoctor have any suggestions.

> There is, though it's going to require abstracting things a little bit more, like get, _getAs and has becoming factories that receive the environment object, so we can call them in both node and deno exports, passing the respective object (process.env for node and Deno.env.toObject() for deno).

I see. Well, I’m in no place to evaluate if that’s worth the time so right now. I wouldn’t want to over complicate this, maybe it’s fine to duplicate the code for now as an experiment. Anyways, it’s your call

**Comment by khaosdoctor** [comment]
I think both implementations should lie in the same place because the changes to the code are minimal, but I agree there should be a reuse of code, this way the only thing we change is the `package.json` to Node and `mod.ts` to Deno, but keep the same code in the backend.

About the API, I like it the way it is, but the only change I'd make is about `env.int` I think everything should be `env.get()` then the conversions, like: `env.get().toInt()`

**Comment by rwillians** [comment]
> I think both implementations should lie in the same place because the changes to the code are minimal, but I agree there should be a reuse of code, this way the only thing we change is the `package.json` to Node and `mod.ts` to Deno, but keep the same code in the backend.

Sounds good, whatever you guys decide I'm go for it.


> About the API, I like it the way it is, but the only change I'd make is about `env.int` I think everything should be `env.get()` then the conversions, like: `env.get().toInt()`

I agree on having separated functions to cast a value, not sure the best way to do it yet in a way that we can keep the API really simple to use. Not sure an `env.get().toInt()` would be possible since the return of `get()` would be string or null, but I see you point.

I have experimented with different APIs for config, this was the last one I was working on before I switch to elixir [@deeptrace/config](https://github.com/deeptrace/js-packages/blob/master/packages/config/test/usage.test.js#L45). The API is harder to use and it's focused on the entire config object (as you might guess from the package's name), but it uses a middleware approach to validate and transform an environment variable's value.

**Pull request closed without merging**